test_data/rules/unrelated_type_equality_checks.dart

```diff
@@ -178,12 +178,26 @@ void function30() {
   if (c.determinant == callable2) print('someFunction30'); // OK
 }
 
+void function31() {
+  var x = EnumOpKind1.delete;
+  if (x == EnumOpKind1.delete) print('delete'); // OK
```

Comment:
Let's add another test comparing two values of the same enum type. `x == EnumOpKind1.update`.